backend/dataall/modules/dataset_sharing/services/share_object_service.py

```diff
@@ -42,12 +44,36 @@
 from dataall.modules.datasets_base.db.dataset_repositories import DatasetRepository
 from dataall.modules.datasets_base.db.dataset_models import DatasetTable, Dataset
 from dataall.modules.datasets_base.services.permissions import DATASET_TABLE_READ
+from dataall.base.aws.iam import IAM
+
 import logging
 
 log = logging.getLogger(__name__)
 
 
 class ShareObjectService:
+    @staticmethod
+    def verify_principal_role(session, share: ShareObject) -> bool:
+        role_name = share.principalIAMRoleName
+        env = EnvironmentService.get_environment_by_uri(session, share.environmentUri)
+        principal_role = IAM.get_role_arn_by_name(account_id=env.AwsAccountId, region=env.region, role_name=role_name)
+        return principal_role is not None
+
+    @staticmethod
+    def update_all_share_items_status(
+        session, shareUri, new_health_status: str, message, previous_health_status: str = None
+    ):
+        for item in ShareObjectRepository.get_all_sharable_items(
+            session, shareUri, healthStatus=previous_health_status
+        ):
```

Comment:
We should only sent to Unhealthy those items that are currently shared. If an item is in PendingState status it should not be in any state yet